kcbq-connector/src/test/java/com/wepay/kafka/connect/bigquery/preprocess/SinkRecordFilterTest.java

```diff
@@ -0,0 +1,198 @@
+package com.wepay.kafka.connect.bigquery.preprocess;
+
+import com.jayway.jsonpath.InvalidPathException;
+import com.wepay.kafka.connect.bigquery.SinkConnectorPropertiesFactory;
+import com.wepay.kafka.connect.bigquery.config.BigQuerySinkConfig;
+import org.apache.log4j.AppenderSkeleton;
+import org.apache.log4j.Logger;
+import org.apache.log4j.spi.LoggingEvent;
+import org.hamcrest.CoreMatchers;
+import org.hamcrest.Matcher;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.Assert.*;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+
+public class SinkRecordFilterTest {
+    private static final String INCLUDE_CONDITION_CONFIG = "includeCondition";
+    private static final String EXCLUDE_CONDITION_CONFIG = "excludeCondition";
+    private static Map testMap;
+    private SinkRecordFilter sinkRecordFilter;
+
+
+    public static SinkRecordFilter initializeSinkRecordFilter(String prop, String condition) {
+        Map<String, String> configProperties = new SinkConnectorPropertiesFactory().getProperties();
+        if (prop != null) {
+            configProperties.put(prop, condition);
+        }
+        return new SinkRecordFilter(new BigQuerySinkConfig(configProperties));
+    }
+
+
+
+    @BeforeClass
+    public static void initializeTestMap() {
+        // The map looks like:
+        // {
+        //     Inventories = [
+        //         {FINANCIAL_BLOCK=Balance Sheet, COUNTRY_ID=8},
+        //         {FINANCIAL_BLOCK=Income Statement, COUNTRY_ID=8}
+        //        ],
+        //     High Performance Mixed Signal ( HPMS ) =
+        //         [{FINANCIAL_BLOCK=Income Statement, COUNTRY_ID=8}]
+        // }
+
+        testMap = new HashMap<>();
+
+        List<Map<String ,Object>> inventoriesList = new ArrayList<>();
+        Map<String ,Object> map1 = new HashMap<>();
+        map1.put("FINANCIAL_BLOCK", "Balance Sheet");
+        map1.put("COUNTRY_ID", 8);
+
+        inventoriesList.add(map1);
+
+
+        Map<String ,Object> map2 = new HashMap<>();
+        map2.put("FINANCIAL_BLOCK", "Income Statement");
+        map2.put("COUNTRY_ID", 8);
+
+        inventoriesList.add(map2);
+
+        testMap.put("Inventories", inventoriesList);
+
+        List<Map<String ,Object>> highPerformanceList = new ArrayList<>();
+        highPerformanceList.add(map2);
+
+        testMap.put("High Performance Mixed Signal ( HPMS )", highPerformanceList);
+
+    }
+
+
+    @Test
+    public void testFilterConstructor() {
```

Comment:
`testFilterConfiguration` or `testCreateFilter` may be a more appropriate name. Whether it is in the constructor or not is less relevant. It's usually better to make the method name explicit in what failure condition or action it is testing.